template.yaml

```diff
@@ -234,6 +260,7 @@ Resources:
               - sts:AssumeRole
       ManagedPolicyArns:
         - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
+      PermissionsBoundary: !If [CreateDefaultBoundaryPolicy, !Ref DefaultPermissionBoundaryPolicy, !Ref PermissionBoundaryPolicyArn]
```

Comment:
one small query!!  
Since we have attached the `PermissionsBoundary` in Globals, so it would be applicable for all the resources right??

Reply:
no, it's only used in Globals for `Function`, and there is [no such section](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-specification-template-anatomy-globals.html) available for roles (for security reasons I think)